Evaluate-ModelarDB-Compression/main.py

```diff
@@ -0,0 +1,281 @@
+""" Script for evaluating ModelarDB with different data sets. """
+
+
+import sys
+import math
+import time
+import signal
+import tempfile
+import subprocess
+
+import numpy
+from pyarrow import parquet
+from pyarrow import flight
+
+
+# Configuration.
+MODELARDB_REPOSITORY = "https://github.com/ModelarData/ModelarDB-RS.git"
+UTILITIES_REPOSITORY = "https://github.com/ModelarData/Utilities.git"
+TABLE_NAME = "evaluate"
+STDOUT = subprocess.PIPE
+STDERR = subprocess.PIPE
+
+
+# Helper Functions.
+def extract_repository_name(url):
+    return url[url.rfind("/") + 1 : url.rfind(".")] + "/"
+
+
+def git_clone(url):
+    subprocess.run(["git", "clone", url], stdout=STDOUT, stderr=STDERR)
+
+
+def cargo_build_release(modelardb_folder):
+    process = subprocess.run(
+        ["cargo", "build", "--release"],
+        cwd=modelardb_folder,
+        stdout=STDOUT,
+        stderr=STDERR,
+    )
+
+    return process.returncode == 0
+
+
+def start_modelardbd(modelardb_folder, data_folder):
+    process = subprocess.Popen(
+        ["target/release/modelardbd", data_folder],
+        cwd=modelardb_folder,
+        stdout=STDOUT,
+        stderr=STDERR,
+    )
+
+    # Ensure process is fully started.
+    while not b"Starting Apache Arrow Flight on" in process.stdout.readline():
+        time.sleep(10)
+
+    return process
+
+
+def errors_occurred(output_stream):
+    normalized = output_stream.lower()
+    return b"error" in normalized or b"panicked" in normalized
+
+
+def print_stream(output_stream):
+    print()
+    print(output_stream.decode("utf-8"))
+
+
+def ingest_test_data(utilities_loader, test_data, error_bound):
+    process = subprocess.run(
+        [
+            "python3",
+            utilities_loader,
+            "127.0.0.1:9999",
+            TABLE_NAME,
+            test_data,
+            error_bound,
+        ],
+        stdout=STDOUT,
+        stderr=STDERR,
+    )
+
+    if errors_occurred(process.stderr):
+        print_stream(process.stderr)
+        return True
+
+
+def retrieve_schema(flight_client):
+    flight_descriptor = flight.FlightDescriptor.for_path(TABLE_NAME)
+    schema_result = flight_client.get_schema(flight_descriptor)
+    return schema_result.schema
+
+
+def retrieve_ingested_column(flight_client, column_name, timestamp_column):
+    ticket = flight.Ticket(
```

Comment:
Maybe use an [f-string](https://peps.python.org/pep-0498/) instead to make it a little more readable.

Reply:
Fixed.